doc/design.md

```diff
@@ -0,0 +1,194 @@
+# Design
+
+
+## Introduction
+This library is an event-driven framework that allows users to build complex 
```

Comment:
We should introduce the framework more generally and then maybe state that we choose a event driven design because of the requirements and the limitation that pquery only is available for Lua at the moment.

PS: A dataflow based programming model/interface for the user is still possible. I will do something similar for machine learning. Where I build an execution graph and have an event driven graph runner which visits in the event handler the next node in the graph und returns its queries. I am only lazy regarding loops, which handle by calling another instance of this framework. However, this should be possible to incorporate as well. We might add this dataflow based programming model to this library, but not now.